### PR TITLE
elizCurtnazar/tema7

### PR DESCRIPTION
**Pull request by curtnazarEliz**

### CODE REVIEW

**Review by pmadalina88 — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -31,6 +31,14 @@
             <artifactId>selenium-java</artifactId>
             <version>4.7.2</version>
         </dependency>
+
```
Comment (95teo) [inline pom.xml:34]: TestNG dependency is already defined in pom.xml file so please remove all of these lines for testng dependency added by you. No need to added multiple times and duplicate the code.

File: src/test/java/elizCurtnazar/tests/tema7/HomeButton.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        homeButton.click();
+        Assert.assertEquals(driver.getCurrentUrl(), "https://katalon-demo-cura.herokuapp.com/");
+
+    }
+
+    @Test
+    public void loginPage() {
+        driver.get("https://katalon-demo-cura.herokuapp.com/profile.php#login");
+        goToTheHomePage();
+
```
Comment (95teo) [inline src/test/java/elizCurtnazar/tests/tema7/HomeButton.java:24]: Please add an assertion at the end of the test case. Thanks!

File: src/test/java/elizCurtnazar/tests/tema7/HomeButton.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    }
+
+    @Test
+    public void historyPage() {
+        driver.get("https://katalon-demo-cura.herokuapp.com/history.php#history");
+        goToTheHomePage();
+    }
```
Comment (95teo) [inline src/test/java/elizCurtnazar/tests/tema7/HomeButton.java:31]: Please add an assertion at the end of the test case. Thanks!

File: src/test/java/elizCurtnazar/tests/tema7/HomeButton.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+    @Test
+    public void profilePage() {
+        driver.get("https://katalon-demo-cura.herokuapp.com/profile.php#profile");
+        goToTheHomePage();
+        Assert.assertNotEquals(driver.getCurrentUrl(), "https://katalon-demo-cura.herokuapp.com/", "!!ERROR!!");
```
Comment (95teo) [inline src/test/java/elizCurtnazar/tests/tema7/HomeButton.java:37]: Here it should be better to use assertEquals considering the assertion you're making.

File: src/test/java/elizCurtnazar/tests/tema7/BaseTest.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    }
+
+    @AfterMethod(groups = "redirect", alwaysRun = true) //alwaysRun = true. metoda mereu da quit, chiar daca e vreo erroare
+    public void quitDriver() {
+        if(driver!=null) {
+            driver.quit();
+            System.out.println("Executed after method");
+        }
+    }
+
```
Comment (95teo) [inline src/test/java/elizCurtnazar/tests/tema7/BaseTest.java:27]: You can remove these 2 blank lines from here.

File: src/test/java/elizCurtnazar/tests/tema7/HomeButton.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
```
Comment (95teo) [inline src/test/java/elizCurtnazar/tests/tema7/HomeButton.java:38]: You can remove these blank lines from here.

File: src/test/java/elizCurtnazar/tests/tema7/HomeButton.java [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+
+    }
+
```
Comment (95teo) [inline src/test/java/elizCurtnazar/tests/tema7/HomeButton.java:42]: You can remove these 2 blank lines from here.

File: src/test/java/elizCurtnazar/tests/tema7/MakeAppointmentScenariosTest.java [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+import org.testng.Assert;
+import org.testng.annotations.Test;
+import org.testng.asserts.SoftAssert;
+
+import java.util.List;
+
+public class MakeAppointmentScenariosTest extends BaseTest{
+
+    @Test
+    public void clickMakeAppointmentButton() {
```
Comment (95teo) [inline src/test/java/elizCurtnazar/tests/tema7/MakeAppointmentScenariosTest.java:15]: You are missing a negative scenario for appointment. From what I can see this is only the positive scenario. Please double check the homework requirement. Thanks!

**Review by 95teo — CHANGES_REQUESTED** [review]

File: src/test/java/elizCurtnazar/tests/tema7/HomeButton.java (hunk 4 above)
Comment (curtnazarEliz) [reply src/test/java/elizCurtnazar/tests/tema7/HomeButton.java:37]: Am crezut ca trebuia facut una sa dea erroare.

File: src/test/java/elizCurtnazar/tests/tema7/BaseTest.java (hunk 5 above)
Comment (curtnazarEliz) [reply src/test/java/elizCurtnazar/tests/tema7/BaseTest.java:27]: done

File: src/test/java/elizCurtnazar/tests/tema7/HomeButton.java (hunk 6 above)
Comment (curtnazarEliz) [reply src/test/java/elizCurtnazar/tests/tema7/HomeButton.java:38]: done

File: src/test/java/elizCurtnazar/tests/tema7/HomeButton.java (hunk 7 above)
Comment (curtnazarEliz) [reply src/test/java/elizCurtnazar/tests/tema7/HomeButton.java:42]: done

File: pom.xml (hunk 1 above)
Comment (curtnazarEliz) [reply pom.xml:34]: am comentat sa- mi spui daca sigur sunt acelea care trebuie sterse

File: src/test/java/elizCurtnazar/tests/tema7/HomeButton.java (hunk 2 above)
Comment (curtnazarEliz) [reply src/test/java/elizCurtnazar/tests/tema7/HomeButton.java:24]: done

File: src/test/java/elizCurtnazar/tests/tema7/HomeButton.java (hunk 3 above)
Comment (curtnazarEliz) [reply src/test/java/elizCurtnazar/tests/tema7/HomeButton.java:31]: done

File: src/test/java/elizCurtnazar/tests/tema7/MakeAppointmentScenariosTest.java (hunk 8 above)
Comment (curtnazarEliz) [reply src/test/java/elizCurtnazar/tests/tema7/MakeAppointmentScenariosTest.java:15]: ok

**Review by sorinfoca — APPROVED** [review]

File: pom.xml [hunk 9]
```diff
@@ -26,11 +26,19 @@
             <artifactId>webdrivermanager</artifactId>
             <version>5.3.1</version>
         </dependency>
-        <dependency>
-            <groupId>org.seleniumhq.selenium</groupId>
-            <artifactId>selenium-java</artifactId>
-            <version>4.7.2</version>
-        </dependency>
+<!--        <dependency>-->
```
Comment (95teo) [inline pom.xml:29]: These changes are not ok. You only needed to remove lines 34 to 41 from this pom file ->
image

File: src/test/java/elizCurtnazar/tests/tema7/HomeButton.java [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+public class HomeButton extends BaseTest {
+
+    @Test
+    public void goToTheHomePage() {
+        WebElement menu = driver.findElement(By.cssSelector(".btn-dark.btn-lg.toggle"));
+        menu.click();
+        WebElement homeButton = driver.findElement(By.xpath("/html/body/nav/ul/li[2]/a"));
+        homeButton.click();
+//        Assert.assertEquals(driver.getCurrentUrl(), "https://katalon-demo-cura.herokuapp.com/");
```
Comment (95teo) [inline src/test/java/elizCurtnazar/tests/tema7/HomeButton.java:16]: Yes, this line should be fine. Please uncomment it.

File: src/test/java/elizCurtnazar/tests/tema7/MakeAppointmentScenariosTest.java [hunk 11]
```diff
@@ -12,7 +12,7 @@
 public class MakeAppointmentScenariosTest extends BaseTest{
 
     @Test
-    public void clickMakeAppointmentButton() {
+    public void clickMakeAppointmentButton(String expectedResult, String actualResults) {
```
Comment (95teo) [inline src/test/java/elizCurtnazar/tests/tema7/MakeAppointmentScenariosTest.java:15]: You are missing the negative scenario for appointment -> 
image

**Review by 95teo — CHANGES_REQUESTED** [review]

**Pull request merged**